docs/reference/repo_properties.rst

```diff
@@ -0,0 +1,162 @@
+
+Package Repository Properties
+*****************************
+
+.. _ppa-properties:
+PPA properties
+==============
+
+The following properties are supported for PPA-type repositories:
+
+- type
+   - Type: enum[string]
+   - Description: Specifies type of package-repository, must currently be
+     ``apt``
+   - Examples: ``type: apt``
+- ppa
+   - Type: string
+   - Description: PPA shortcut string
+   - Format: <ppa-owner>/<ppa-name>
+   - Examples:
+      - ``ppa: snappy-devs/snapcraft-daily``
+      - ``ppa: mozillateam/firefox-next``
+
+
+.. _deb-properties:
+Deb properties
+==============
+
+The following properties are supported for Deb-type repositories:
+
+- architectures
+   - Type: list[string]
+   - Description: Architectures to enable, or restrict to, for this repository
+   - Default: If unspecified, architectures is assumed to match the host’s
+     architecture
+   - Examples:
+      - ``architectures: [i386]``
+      - ``architectures: [i386, amd64]``
+- components
+   - Type: list[string]
+   - Description: Apt repository components to enable: e.g.
+     ``main``, ``multiverse``, ``unstable``
+   - Examples:
+       - ``components: [main]``
+       - ``components: [main, multiverse, universe, restricted]``
+- formats
+   - Type: list[string]
+   - Description: List of deb types to enable
+   - Default: If unspecified, format is assumed to be ``deb``, i.e. ``[deb]``
+   - Examples:
+       - ``formats: [deb]``
+       - ``formats: [deb, deb-src]``
+- key-id
+   - Type: string
+   - Description: 40 character GPG key identifier ("long-form thumbprint" or
+     "fingerprint")
+     If not using a key-server, Snapcraft will look for the corresponding key
+     at: ``<project>/snap/keys/<key-id[-8:]>.asc``. To determine a key-id from a
```

Comment:
How do you not use a keyserver? Set `key-server` to empty?

Reply:
Good question; I reworked this paragraph to clarify that the Application will always look for the key locally first, and then fallback to the key-server (with its default fallback)
@sergiusens one issue here is that this still references `snap/keys/...`, which will only apply to Snapcraft. How can we: a) document this "generically" and b) support the equivalent in Rockcraft, when the time comes?